jsettlers.logic/src/jsettlers/logic/buildings/spawn/SpawnBuilding.java

```diff
@@ -45,11 +44,11 @@ public final EMapObjectType getFlagType() {
 
 	@Override
 	protected int constructionFinishedEvent() {
-		return PRODUCE_PERIOD;
+		return subTimerEvent();
```

Comment:
Maybe we should leve the 2 seconds period here. By this we ensure the first movable is spawned quickly after the building is finished, but after all builders really finished there work. I know this is a bit of a hack to compensate the non-ideal implementation of the builders but it could also look nice ;)